packages/adapters-library/src/scripts/deleteAdapterMetadata.ts

```diff
@@ -0,0 +1,207 @@
+import fs from 'node:fs'
+import path from 'node:path'
+import Database, { Database as DatabaseType } from 'better-sqlite3'
+import chalk from 'chalk'
+import { Command } from 'commander'
+import { Protocol } from '../adapters/protocols'
+import { supportedProtocols } from '../adapters/supportedProtocols'
+import { AdaptersController } from '../core/adaptersController'
+import { ZERO_ADDRESS } from '../core/constants/ZERO_ADDRESS'
+import { Chain, ChainIdToChainNameMap } from '../core/constants/chains'
+import { ProviderMissingError } from '../core/errors/errors'
+import { CustomJsonRpcProvider } from '../core/provider/CustomJsonRpcProvider'
+import { filterMapSync } from '../core/utils/filters'
+import { logger } from '../core/utils/logger'
+import {
+  AdditionalMetadataWithReservedFields,
+  Erc20ExtendedMetadata,
+  IProtocolAdapter,
+  ProtocolToken,
+} from '../types/IProtocolAdapter'
+import { Erc20Metadata } from '../types/erc20Metadata'
+import { getInvalidAddresses } from './addressValidation'
+import { multiChainFilter, multiProtocolFilter } from './commandFilters'
+
+export function deleteAdapterMetadata(
+  program: Command,
+  chainProviders: Record<Chain, CustomJsonRpcProvider>,
+  adaptersController: AdaptersController,
+) {
+  program
+    .command('delete-adapter-metadata')
+    .option(
+      '-p, --protocols <protocols>',
+      'protocol id (e.g. stargate or aave-v2)',
+    )
+    .option(
+      '-pd, --products <products>',
+      'product id (e.g. voting-escrow or market-borrow)',
+    )
+    .option(
+      '-c, --chains <chains>',
+      'chains id or name (e.g. ethereum,arbitrum,linea)',
+    )
+    .showHelpAfterError()
+    .action(
+      async ({
+        protocols,
+        products,
+        chains,
+      }: { protocols: string; products: string; chains: string }) => {
+        console.log('Starting deletion process...')
+        console.log('Protocols:', protocols)
+        console.log('Products:', products)
+        console.log('Chains:', chains)
+
+        const filterProtocolIds = multiProtocolFilter(protocols)!
+        if (
+          !protocols ||
+          (filterProtocolIds && filterProtocolIds.length !== 1)
+        ) {
+          throw new Error('One protocol must be supplied at a time')
+        }
+        const protocolId = filterProtocolIds.pop()!
+
+        const filterProductIds = products?.split(',')
+        if (!products || (filterProductIds && filterProductIds.length !== 1)) {
+          throw new Error('One product must be supplied at a time')
+        }
+        const productId = filterProductIds?.pop()!
+
+        const filterChainIds = multiChainFilter(chains)!
+        if (!chains || (filterChainIds && filterChainIds.length !== 1)) {
+          throw new Error('One chain must be supplied at a time')
+        }
+        const chainId = filterChainIds.pop()!
+
+        const db = new Database(`./${ChainIdToChainNameMap[chainId]}.db`, {
+          verbose: console.log,
+        })
+
```

Comment:
We could run the query to fetch the pools once and passed the value to every delete query, instead of calculating it every time.

```
SELECT p.pool_id
      FROM adapters a
      JOIN pools p ON p.adapter_id = a.adapter_id
      WHERE a.protocol_id = ? AND a.product_id = ?
```

The way it's written was ok for a sql script, where you cannot (should not) use variables, but that restriction does not apply here.